cloudinit/net/activators.py

```diff
@@ -255,8 +255,8 @@ def bring_down_interface(device_name: str) -> bool:
 # version to encompass both seems overkill at this point
 DEFAULT_PRIORITY = [
     IfUpDownActivator,
-    NetworkManagerActivator,
     NetplanActivator,
+    NetworkManagerActivator,
     NetworkdActivator,
```

Comment:
I don't think this is necessary when you've changed it to check if the `nmcli` tool exists.

Reply:
Thanks for the review comments and previous PR too!

The presence of nmcli (just as the presence of systemd-networkd) doesn't mean that cloud-init should primarily use those tools directly to bring up specific interfaces. Given that netplan renders configuration aritfacts to either network backend it should be preferred because Images which have netplan.io installed have made an opinionated decision to install netplan to manage network config.  It feels a bit like undercutting that decision by going directly to the network backend tools even when netplan is installed and will likely result in more cases of misconfiguration as some configuraiton defined up in /etc/netplan/* config may have defined a separate `renderer: sytemd-networkd` configuration which nmcli will know nothing about.

Reply:
That makes sense. I have also adjusted my PR (#1435) to now order Netplan at the highest for that reason.